### PR TITLE
[JENKINS-36076] Acceptance tests for External Workspace Manager Plugin

### PR DESCRIPTION
**Pull request by alexsomai**
Related to issue [[JENKINS-36076]](https://issues.jenkins-ci.org/browse/JENKINS-36076)

Summary of this pull request:
 - [x] Write acceptance tests for External Workspace Manager Plugin

CC @oleg-nenashev @martinda

### CODE REVIEW

**Comment by alexsomai** [comment]
Most probably the test will fail while running remote on Jenkins, because I'm installing External Workspace Manager Plugin from local .hpi artifact.
This is until the stable version will be released.
It passes locally.

File: src/main/java/org/jenkinsci/test/acceptance/plugins/external_workspace_manager/ExternalGlobalConfig.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package org.jenkinsci.test.acceptance.plugins.external_workspace_manager;
+
+import org.jenkinsci.test.acceptance.po.PageAreaImpl;
+import org.jenkinsci.test.acceptance.po.PageObject;
+
+/**
+ * Helper class for interacting with External lWorkspace Manager Plugin global config page.
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/external_workspace_manager/ExternalGlobalConfig.java]: typo: lWorkspace

File: src/main/java/org/jenkinsci/test/acceptance/plugins/external_workspace_manager/ExternalNodeConfig.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package org.jenkinsci.test.acceptance.plugins.external_workspace_manager;
+
+import org.jenkinsci.test.acceptance.po.PageAreaImpl;
+import org.jenkinsci.test.acceptance.po.PageObject;
+
+/**
+ * Helper class for interacting with External lWorkspace Manager Plugin node config page.
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/external_workspace_manager/ExternalNodeConfig.java]: same

**Comment by oleg-nenashev** [comment]
The tests look good to me.
As we discussed, it makes sense to add tests demonstrating integration with ws-cleanup and run-selector plugin.

**Comment by alexsomai** [comment]
I've added test for ws-cleanup integration.
The run-selector integration is tested in `shareWorkspaceTwoJobsTwoNodes`

Do not merge this until I release the stable version

File: src/main/java/org/jenkinsci/test/acceptance/plugins/external_workspace_manager/ExternalGlobalConfig.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.jenkinsci.test.acceptance.po.PageObject;
+
+/**
+ * Helper class for interacting with External Workspace Manager Plugin global config page.
+ *
+ * @author Alexandru Somai
+ */
+public class ExternalGlobalConfig extends PageAreaImpl {
+
+    public ExternalGlobalConfig(PageObject context) {
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/external_workspace_manager/ExternalGlobalConfig.java]: It is a good practice to require the exact type of PageObject in PageArea so it is obvious to the client how to use it.

File: src/main/java/org/jenkinsci/test/acceptance/plugins/external_workspace_manager/ExternalNodeConfig.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import org.jenkinsci.test.acceptance.po.PageObject;
+
+/**
+ * Helper class for interacting with External Workspace Manager Plugin node config page.
+ *
+ * @author Alexandru Somai
+ */
+public class ExternalNodeConfig extends PageAreaImpl {
+
+    public ExternalNodeConfig(PageObject context) {
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/external_workspace_manager/ExternalNodeConfig.java]: It is a good practice to require the exact type of PageObject in PageArea so it is obvious to the client how to use it.

**Comment by oleg-nenashev** [comment]
Looks good 👍 . Some additional data structure checks may be useful, but it's fine for ATH spotchecks

**Comment by oleg-nenashev** [comment]
Test failures are unrelated

**Comment by oleg-nenashev** [comment]
Merging the current state since the plugin has been already released

**Pull request merged**